### DISCUSSION TITLE
What is the reason to replace existing rule with new rule when longer?

### DISCUSSION THREAD

**Discussion by eddyw** [mattpocock/ts-error-translator#13] [Q&A] [answered]
Hi 👋 

I'm failing to understand this particular piece of code here:
```ts
          if (errorMessageByKey[key]) {
            const existingRule = errorMessageByKey[key];

            /**
             * If the new rule is longer than the existing rule,
             * replace the existing rule with the new rule.
             */
            errorMessageByKey[key] =
              newError.length > existingRule.error.length
                ? errorObj
                : existingRule;
          } else {
            errorMessageByKey[key] = errorObj;
          }
```
What is the intention?
Is there an example diagnostic? I'm failing to reproduce it and it's driving me crazy 😄

**Comment by mattpocock** [comment, marked as answer]
The reason is that some errors are actually a subset of other errors - i.e:

"{0} is bad"
"{0} is bad because of something"

In that case, we want to choose the longer version of the same rule.

HOWEVER - this code might not be needed any more, we should add tests in this area to check it out.

**Reply by eddyw** [reply]
Oh that's interesting 👍
